### PR TITLE
Add back pylint as info-only for core

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: core/nox.py [hunk 1]
```diff
@@ -43,15 +41,21 @@ def unit_tests(session, python_version):
...
-    session.install('flake8')
+    session.install('flake8', 'pylint', 'gcp-devrel-py-tools')
     session.install('.')
     session.run('flake8', 'google/cloud/core')
+    session.run(
+        'gcp-devrel-py-tools', 'run-pylint',
+        '--library-filesets', 'google',
+        '--test-filesets', 'tests',
+        # Temporarily allow this to fail.
+        success_codes=range(0, 100))
```
Comment (dhermes) [inline core/nox.py]: You probably want `list(range(0, 100))`? If the script is invoked by Python 3, this will be an iterator.

File: core/nox.py [hunk 2]
```diff
@@ -43,15 +41,21 @@ def unit_tests(session, python_version):
...
+    Returns a failure if the linters find linting errors or sufficiently
     serious code quality issues.
     """
     session.interpreter = 'python3.6'
-    session.install('flake8')
+    session.install('flake8', 'pylint', 'gcp-devrel-py-tools')
     session.install('.')
     session.run('flake8', 'google/cloud/core')
+    session.run(
+        'gcp-devrel-py-tools', 'run-pylint',
```
Comment (dhermes) [inline core/nox.py]: Can it somehow be less opaque how `pylintrc` and `pylintrc.test` are generated by `gcp-devrel-py-tools`?

Comment (jonparrott) [inline core/nox.py]: What do you recommend?

File: core/nox.py (hunk 1 above)
Comment (jonparrott) [inline core/nox.py]: It works fine in python 3.

File: core/nox.py (hunk 2 above)
Comment (dhermes) [inline core/nox.py]: Well first, could you explain how it works, then we can go from there?

File: core/nox.py (hunk 1 above)
Comment (dhermes) [inline core/nox.py]: OK cool. TIL: `__contains__` works as one might expect on a `range` object Python 3

File: core/nox.py (hunk 2 above)
Comment (jonparrott) [inline core/nox.py]: It's basically what your old "run_pylint" script does, however, the key difference here is that we could specify a `pylintrc.py` file here and use it to programmatically override the defaults.

https://github.com/GoogleCloudPlatform/python-repo-tools/blob/master/gcp_devrel/tools/pylint.py

Comment (dhermes) [inline core/nox.py]: Ohhh that's fancy!

Comment (jonparrott) [inline core/nox.py]: So WDYT, is there a comment or anything I can put here to make things easier?

Comment (dhermes) [inline core/nox.py]: I think we should just have a `pylintrc.py` file, even if it is very/totally bare. Then you can just make a comment referring to it?

**Comment by jonparrott** [comment]
@dhermes added a "empty" config file for you. PTAL?

File: core/pylint.config.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# Library configuration
+
+# library_additions = {}
+# library_replacements = {}
+
+# Test configuration
+
+# test_additions = copy.deepcopy(library_additions)
+# test_replacements = copy.deepcopy(library_replacements)
```
Comment (dhermes) [inline core/pylint.config.py]: Why comment these out? Does it break something to have them empty?

Comment (jonparrott) [inline core/pylint.config.py]: Yep.

Comment (dhermes) [inline core/pylint.config.py]: OK.

**Comment by dhermes** [comment]
This LGTM.

**Comment by jonparrott** [comment]
Cool, I'll send another PR to add this to all the other packages.

**Comment by dhermes** [comment]
SGTM. Do we think @lukesneeringer cares? (Prepare for some lint violations since we've had it turned off.)

**Comment by jonparrott** [comment]
It's in info-only mode so it will not cause any new failures. Luke gave
approval in the email chain you were on.

**Comment by dhermes** [comment]
Boom roasted

**Pull request merged**